addons/mrp/models/mrp_workorder.py

```diff
@@ -206,6 +207,13 @@ def _compute_allowed_lots_domain(self):
                     qties_done_per_lot[finished_workorder_line.lot_id.id].append(finished_workorder_line.qty_done)
 
             qty_to_produce = production.product_qty
+            if production.product_id.tracking == 'serial' and production.product_uom_id.uom_type != 'reference':
+                reference_uom = self.env['uom.uom'].search([('category_id', '=', production.product_uom_id.category_id.id), ('uom_type', '=', 'reference')])
```

Comment:
we want to convert into the product_id UoM. It's not always the same as the reference